draft-ietf-quic-multipath.md

```diff
@@ -1227,11 +1322,153 @@ PATH_AVAILABLE frame is considered lost, the peer SHOULD resend the frame
+their Path Identifier is 0. Also the retire prior to field of NEW_CONNECTION_ID frames
+just effect the Connection IDs of initial path with path ID 0. This machanism 
+is compatible with {{QUIC-Transport}}.
+
+
+## MP_RETIRE_CONNECTION_ID frames {#mp-retire-conn-id-frame}
+
+An endpoint sends a MP_RETIRE_CONNECTION_ID frame (type=0x15228c0a) instead of 
+RETIRE_CONNECTION_ID frame to indicate that it will no longer use a connection ID 
+that was issued by its peer. This includes the connection ID provided during the handshake. 
+Sending a MP_RETIRE_CONNECTION_ID frame also serves as a request to the peer 
+to send additional connection IDs for future use, unless the path specified 
+by Path Identifier has been abandoned. New connection IDs can be 
+delivered to a peer using the MP_NEW_CONNECTION_ID frame ({{mp-new-conn-id-frame}}).
+
+Retiring a connection ID invalidates the stateless reset token associated with that connection ID.
+
+MP_RETIRE_CONNECTION_ID frames are formatted as shown in {{fig-mp-retire-connection-id-frame-format}}.
+
+~~~
+MP_RETIRE_CONNECTION_ID Frame {
+  Type (i) = 0x15228c0a,
+  Path Identifier (i),
+  Sequence Number (i),
+}
+~~~
+{: #fig-mp-retire-connection-id-frame-format title="MP_RETIRE_CONNECTION_ID Frame Format"}
+
+Path Identifier:
+: A path identifier which is pre allocated when the Connection ID is generated, which
+means the current Connection ID can only be used on the corresponding path.
+
+Sequence Number:
+The sequence number assigned to the connection ID by the sender on the path 
+specified by Path Identifier, encoded as a variable-length integer. 
+
+
+## MAX_PATHS frames {#max-paths-frame}
+
+A MAX_PATHS frame (type=0x15228c0b) informs the peer of the cumulative number of paths 
+it is permitted to open. 
+
+MAX_PATHS frames are formatted as shown in {{fig-max-paths-frame-format}}.
+
+~~~
+MAX_PATHS Frame {
+  Type (i) = 0x15228c0b,
+  Maximum Paths (i),
+}
+~~~
+{: #fig-max-paths-frame-format title="MAX_PATHS Frame Format"}
+
+MAX_PATHS frames contain the following field:
+
+Maximum Path Identifier:
+: A count of the cumulative number of path that can be opened 
+over the lifetime of the connection. This value cannot exceed 2^32-1, as it is not 
+possible to encode Path IDs larger than 2^32-1. Receipt of a frame that permits 
+opening of a path with Path Identifier larger than this limit MUST be treated 
+as a connection error of type FRAME_ENCODING_ERROR.
+
+Loss or reordering can cause an endpoint to receive a MAX_PATHS frame with 
+a lower path limit than was previously received. MAX_PATHS frames that 
+do not increase the path limit MUST be ignored.
+
+An endpoint MUST NOT open more paths than permitted by the current path limit 
+set by its peer. For instance, a server that receives a path limit of 3 
+is permitted to open paths 0, 1, 2, but not path 3. An endpoint MUST terminate 
+a connection with an error of type MP_PROTOCOL_VIOLATION if a peer opens 
+more paths than was permitted. 
```

Comment:
So coming back to the discussion about the MAX_PATHS frame, I pretty like the idea that it should indicate the maximum Path ID value that could be used from the sender's perspective (to keep the path management asymmetry where e.g., a client may want to open many paths, but the server wants to keep resources up to x paths). I would probably rename the "Maximum paths" field of the MAX_PATHS frame to "Maximum Path ID", so if a MAX_PATHS {"Maximum Path ID": 3} is received, it should be possible to open paths 0, 1, 2 and 3. This is mainly to limit the path IDs where MP_NEW_CONNECTION_ID frames could be received. But if the endpoint does not send MP_NEW_CONNECTION_ID frame for a given Path ID, the path cannot be opened by its peer anyway (no associated CID). This has also the nice property that endpoints could negotiate multipath with the `initial_max_paths` TP, initially with only one path allowed (if `initial_max_paths` is 0, as the initial path ID is 0; absence of this TP means multipath is not supported).

A last point: I think we should decide whether the multipath extension leaves the door open to server-initiated paths in a latter extension. If this is the case, we should split the Path ID space in even (client-initiated) and odd (server-initiated) values, as done in e.g. stream IDs. For simplicity, the current extension could only allow client-initiated paths (so only even Path IDs).

Reply:
@qdeconinck I'm not sure if any action is needed here but if so, can you open one (or two) new issues?

Otherwise, can I resolve this comment?

Reply:
I think my main concern is, on the long term, to enable server-initiated paths, requiring to split the Path ID space from client-initiated paths to server-initiated ones. #294 seems to address this, so we can resolve this thread for now.

Reply:
okay, please feel free to add further comments to #294 then. I will mark it as resolved for now!